### PR TITLE
selective test (#10)

### PR DESCRIPTION
**Pull request by rscohn2**
Bring back selective testing: https://github.com/oneapi-src/oneMKL/pull/509

Fixes issues:

* Selective test on PR, full test on merge to develop
* Filters *.md, docs, and third-party-programs. I did not filter on scripts because it isn't easy to know if a script affects build
* Restructuring for testing: `node .github/scripts/domain-check.js` to run tests

### CODE REVIEW

File: .github/scripts/domain-check.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    }
+    const files = await prFiles(github, context);
+    const match = matchesPattern(domain, files);
+    console.log("Domain: ", domain)
+    console.log("PR files: ", files);
+    console.log("Match: ", match);
+    return match;
+}
+
+test_patterns = [
```
Comment (dnhsieh-intel) [inline .github/scripts/domain-check.js:63]: Does it make sense to add a comment indicating that this part is to test the javascript itself?

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -21,34 +21,40 @@ jobs:
...
     name: unit tests ${{ matrix.config }} CPU
     steps:
     - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+    - name: Check if the changes affect this domain
+      id: domain_check
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      with:
+        script: |
+         const domainCheck = require('.github/scripts/domain-check.js')
+         return domainCheck({github, context, domain: "${{ matrix.domain}}"})
```
Comment (dnhsieh-intel) [inline .github/workflows/pr.yml]: nitpicking
```suggestion
         return domainCheck({github, context, domain: "${{ matrix.domain }}"})
```

**Review by dnhsieh-intel — APPROVED** [review]
Looks good to me. Thank you! I only have some minor comments.

File: .github/scripts/domain-check.js (hunk 1 above)
Comment (rscohn2) [reply .github/scripts/domain-check.js:63]: Yes. Added comments.

File: .github/workflows/pr.yml (hunk 2 above)
Comment (rscohn2) [reply .github/workflows/pr.yml]: changed

**Review by Rbiessy — APPROVED** [review]
I'm not familiar with javascript but this solution looks good to me and simple enough for anyone to adapt.

**Pull request merged**